src/VisualStudio/Core/Def/Options/PackageSettingsPersisterProvider.cs

```diff
@@ -9,35 +9,24 @@
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Options;
-using IAsyncServiceProvider = Microsoft.VisualStudio.Shell.IAsyncServiceProvider;
-using SAsyncServiceProvider = Microsoft.VisualStudio.Shell.Interop.SAsyncServiceProvider;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
 
-namespace Microsoft.VisualStudio.LanguageServices.Options
-{
-    [Export(typeof(IOptionPersisterProvider))]
-    internal sealed class PackageSettingsPersisterProvider : IOptionPersisterProvider
-    {
-        private readonly IThreadingContext _threadingContext;
-        private readonly IAsyncServiceProvider _serviceProvider;
-        private readonly IGlobalOptionService _optionService;
-        private PackageSettingsPersister? _lazyPersister;
+namespace Microsoft.VisualStudio.LanguageServices.Options;
 
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public PackageSettingsPersisterProvider(
-            IThreadingContext threadingContext,
-            [Import(typeof(SAsyncServiceProvider))] IAsyncServiceProvider serviceProvider,
-            IGlobalOptionService optionService)
-        {
-            _threadingContext = threadingContext;
-            _serviceProvider = serviceProvider;
-            _optionService = optionService;
-        }
+[Export(typeof(IOptionPersisterProvider))]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class PackageSettingsPersisterProvider(
+    IThreadingContext threadingContext,
+    [Import(typeof(SAsyncServiceProvider))] IAsyncServiceProvider serviceProvider,
+    IGlobalOptionService optionService) : IOptionPersisterProvider
+{
+    private PackageSettingsPersister? _lazyPersister;
 
-        public ValueTask<IOptionPersister> GetOrCreatePersisterAsync(CancellationToken cancellationToken)
-        {
-            _lazyPersister ??= new PackageSettingsPersister(_threadingContext, _serviceProvider, _optionService);
-            return new ValueTask<IOptionPersister>(_lazyPersister);
-        }
+    public ValueTask<IOptionPersister> GetOrCreatePersisterAsync(CancellationToken cancellationToken)
+    {
+        _lazyPersister ??= new PackageSettingsPersister(threadingContext, serviceProvider, optionService);
```

Comment:
📝 These are now breaking naming style conventions, and explicit fields should be added back. I'll start working on an analyzer to disallow direct references to a captured primary constructor parameter to avoid this in the future.

Reply:
I would reject that analyzer. This is written the normal way the ldm expects with primary constructors and parameters. 

The naming is correct as well. These are parameters and as such are named as parameters.

Reply:
Yeah, this is now idiomatic C# code. (I'm not a huge fan of it personally, but it _is_ idiomatic.)

Reply:
i find this much easier to read and reason about.  That the entire type is now like 5 lines total, and i can clearly see the constructor parameters flow into this method makes it much clearer to me than having to do all the tracking of parameter to assignment to field to reference.

Reply:
> but it is idiomatic

I would also be OK with the compiler layer policy of just blocking a new feature that isn't compatible with general style we've had.

Reply:
This is compatible with the style we've always had.  It was intentionally designed that way.  The style for parameters and captures has been established for the team for years, and this entirely follows that design.